kv/kvclient/kvstreamer: TestStreamerVaryingResponseSizes failed

kv/kvclient/kvstreamer.TestStreamerVaryingResponseSizes [failed](https://tanzanite.cluster.engflow.com/invocation/485553d2-82cc-4eb4-865f-1928c5443d65) on master @ [e174990dd5015a38b1e9bc6723f270dbe647c3a3](https://github.com/cockroachdb/cockroach/commits/e174990dd5015a38b1e9bc6723f270dbe647c3a3):

Fatal error:

```
panic: test timed out after 4m57s
running tests:
	TestStreamerVaryingResponseSizes (3m55s)

```
Stack: 

```
goroutine 81476 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2259 +0x3b9
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x2d
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestStreamerVaryingResponseSizes
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestStreamerVaryingResponseSizes703775358
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:159: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
```
</p>
</details>
<p>Parameters: <code>attempt=1</code>
, <code>run=30</code>
, <code>shard=1</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStreamerVaryingResponseSizes.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-33360

--- a/pkg/kv/kvclient/kvstreamer/streamer_test.go
+++ b/pkg/kv/kvclient/kvstreamer/streamer_test.go
@@ -589,7 +589,7 @@ ALTER TABLE t SPLIT AT SELECT generate_series(1, 30000, 3000);
 	kvGRPCCallsRegex := regexp.MustCompile(`KV gRPC calls: (\d+,)`)
 	for inOrder := range []bool{false, true} {
 		runner.Exec(t, `SET streamer_always_maintain_ordering = $1;`, inOrder)
-		for i := 0; i < 5; i++ {
+		for i := 0; i < 2; i++ {
 			var gRPCCalls int
 			var err error
 			rows := runner.QueryStr(t, `EXPLAIN ANALYZE SELECT length(blob) FROM t@t_v_idx WHERE v = '1';`)

